docs/ops/cli.md

```diff
@@ -170,29 +170,13 @@ These examples about how to manage a job in CLI.
 
         ./bin/flink cancel <jobID>
 
--   Cancel a job with a savepoint:
+-   Cancel a job with a savepoint (deprecated; use "stop" instead):
```

Comment:
suggest to change to `use "stop -s" command instead`

Reply:
"stop -s" and "stop" both do the same thing. "-s" alone does not do anything, that's why I changed it to "-p" instead to make it clearer that we are actually only passing the savepoint path.

Reply:
I see, but still I'd suggest to explicitly use `stop -p` here so we could prevent change again after supporting normal stop. Wdyt?

Reply:
I would say that @carp84 has a point here. Irrespectively of which option we keep, `-p` or `-s`, I think it makes sense here to propose "use `stop -s/p <location>` so that if the semantics of `stop` or `stop -s/p` (with no further argument) change, the user scripts do not.

Reply:
I think at this point it is not yet clear, where we will converge to in term of FLIP-45/47, so we should try to make as little assumptions about the future features as possible while still keeping the API flexible. My basic assumption is that 

* `stop` always indicates a graceful shutdown
* `cancel` indicates an ungraceful, fast shutdown

I understand we are basically discussing the following two options:

Option 1: `stop [-p savepoint-path] [-d]` (`[]` indicates optional arguments)

If we want to support multiple snapshot formats during graceful shutdowns in the future, we evolve it to:  

`stop [-p snapshot-path] [-d] [-f format]`, while the format defaults to a "savepoint format"`

If we also want to support a graceful shutdown via `stop` without a snapshot at all, we could for example add an additional "format" indicating  "no snapshot" or add an additional flag, which is mutually exclusive to `-f`. 

Option 2: `stop [-s] [-p savepoint-path] [-d]`

The `-s` would indicate that a snapshot should be taken. If `stop` is called without `-s`, we throw an exception in Flink 1.9 (we would need to change the current behavior for this).  

The advantage would be that, if we later want to add a `stop` without a snapshot, it would be more natural in terms of the API.


Personally, I don't see the need for a graceful shutdown without some kind of snapshot (yet). So, I think, Option 1 is a better trade off between flexibility and simplicity for current users.

@carp84 @aljoscha @kl0u: What do you think?

Reply:
I totally agree that a graceful shutdown should be accompanied by some kind of "snapshot" and Option 1 is better, but focusing on this line in the doc, I suggest to explicitly indicate that `cancel with savepoint` could only be replaced by `stop with savepoint`, which will be correct no matter where FLIP-45/47 goes.

It's true that in current master `stop` has the same effect with `stop with savepoint`, but later on this may change. If it did change and now we write the doc as `use "stop" instead`, we still need to change it to `use "stop -p" instead` later on, and if we forgot to, this will mislead our users. What do you think? @knaufk

Reply:
After some offline discussion, I agree we keep the doc as is and do necessary refinement later when FLIP-45/47 is done.